root/programs/MultiPurposeAuthSite/MultiPurposeAuthSite/Controllers/AccountController.cs

```diff
@@ -0,0 +1,2221 @@
+                            {
+                                // サインアップ済み → 外部ログイン追加だけで済む
+
+                                // 外部ログイン（ = UserLoginInfo ）の追加
+                                if (ASPNETIdentityConfig.RequireUniqueEmail)
+                                {
+                                    result = await UserManager.AddLoginAsync(user.Id, externalLoginInfo.Login);
+                                }
+                                else
+                                {
+                                    if (email == user.Email)
+                                    {
+                                        // メアドも一致
+                                        result = await UserManager.AddLoginAsync(user.Id, externalLoginInfo.Login);
+                                    }
+                                    else
+                                    {
+                                        // メアド不一致
+                                        result = new IdentityResult();
+                                    }
+                                }
+
+                                // クレーム（emailClaim, nameClaim, etc.）の追加
+                                if (result.Succeeded)
+                                {
+                                    result = await UserManager.AddClaimAsync(user.Id, emailClaim);
+                                    result = await UserManager.AddClaimAsync(user.Id, nameClaim);
+                                    // ・・・
+                                    // ・・・
+                                    // ・・・
+                                }
+
+                                // 上記の結果の確認
+                                if (result.Succeeded)
+                                {
+                                    // SignInAsync、ExternalSignInAsync
+                                    // 通常のサインイン（外部ログイン「追加」時はSignInAsyncを使用する）
+                                    await SignInManager.SignInAsync(
+                                        user,
+                                        isPersistent: false,    // rememberMe は false 固定（外部ログインの場合）
+                                        rememberBrowser: true); // rememberBrowser は true 固定
+
+                                    //// この外部ログイン・プロバイダでサインイン
+                                    //signInStatus = await SignInManager.ExternalSignInAsync(
+
+                                    // AppScan指摘の反映
+                                    this.FxSessionAbandon();
+                                    // SessionIDの切換にはこのコードが必要である模様。
+                                    // https://support.microsoft.com/ja-jp/help/899918/how-and-why-session-ids-are-reused-in-asp-net
+                                    Response.Cookies.Add(new HttpCookie("ASP.NET_SessionId", ""));
+
+                                    // イベント・ログ出力
+                                    Logging.MyOperationTrace(string.Format("{0}({1}) did ext sign in.", user.Id, user.UserName));
```

Comment:
@daisukenishino 

did ext sign in -> has signed in with a verified external account